printrun/gcoder.py

```diff
@@ -548,9 +571,9 @@ def _preprocess(self, lines = None, build_layers = False,
 
                     # Compute duration
                     if line.command == "G0" or line.command == "G1":
-                        x = line.x if line.x is not None else lastx
-                        y = line.y if line.y is not None else lasty
-                        z = line.z if line.z is not None else lastz
+                        x = line.x if line.x is not None else lastx or 0
```

Comment:
the `or 0` is meant to happen if `lastx` is `None`? In that case I'd suggest (lastx or 0) to make it clear that this is the intention, because the precedence interaction rules of trinary expressions and logical expressions are not immediately obvious. Same for the two lines below.